application/helpers/api_helper.php

```diff
@@ -326,6 +326,11 @@ function filter_remote_url($url, $allowed_schemes = array('http', 'https')) {
         {
             $url = false;
         }
+        /* This blocks INT, including IP in hex format such as 0x7f000001 */
+        if (filter_var($host, FILTER_VALIDATE_INT, FILTER_FLAG_ALLOW_HEX))
+        {
+            $url = false;
+        }
         /*We should check if it is a hostname - resolving url*/
         else
         {
```

Comment:
Why do we still let unresolved URL pass the validation?

Reply:
This predates me, but it looks like it's so that the function has just one "return" which is probably easier for testing.

Reply:
I think this check is for valid, legit DNS names with private IPs. Two cases here, a) an internal host e.g. database.internal.gsa.gov or b) an attacker controls the DNS name and it [resolves to a private address](https://github.com/GSA/project-open-data-dashboard/blob/96bcfad3b4676ec3773904b000ffb2412660488a/tests/validation.bats#L111).